src/application/Component.mjs

```diff
@@ -445,7 +445,10 @@ export default class Component extends Element {
             throw new Error("Ancestor event name must be prefixed by dollar sign.");
         }
 
-        return this._doFireAncestors(name, args);
+        const parent = this._getParentSignalHandler();
+        if(parent){
```

Comment:
style: missing space after if and before bracket.

Reply:
this now has been changed